bff-service/src/gateway-cache.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
```

Comment:
Just FYI: you can use CacheModule, provided by nestjs, more info u can find in https://docs.nestjs.com/techniques/caching#auto-caching-responses

Reply:
Will look into it. Thanks!